packages/core/src/sync-historical/service.ts

```diff
@@ -529,164 +545,231 @@ export class HistoricalSyncService extends Emittery<HistoricalSyncEvents> {
           this.network.name
         })`,
       });
-    };
+    }
+  };
 
-    const queue = createQueue<HistoricalSyncTask>({
-      worker,
-      options: {
-        concurrency: this.network.maxHistoricalTaskConcurrency,
-        autoStart: false,
-      },
-      onError: ({ error, task, queue }) => {
-        const message = getErrorMessage(error);
-        switch (task.kind) {
-          case "LOG_FILTER": {
-            this.common.logger.warn({
-              service: "historical",
-              msg: `Log filter task failed, retrying... [${task.fromBlock}, ${task.toBlock}] (contract=${task.logFilter.contractName}, network=${this.network.name}, error=${message})`,
-            });
-            const priority = Number.MAX_SAFE_INTEGER - task.fromBlock;
-            queue.addTask(task, { priority, retry: true });
-            break;
-          }
-          case "FACTORY_CHILD_ADDRESS": {
-            this.common.logger.warn({
-              service: "historical",
-              msg: `Factory child address task failed, retrying... [${task.fromBlock}, ${task.toBlock}] (contract=${task.factory.contractName}, network=${this.network.name}, error=${message})`,
-            });
-            const priority = Number.MAX_SAFE_INTEGER - task.fromBlock;
-            queue.addTask(task, { priority, retry: true });
-            break;
-          }
-          case "FACTORY_LOG_FILTER": {
-            this.common.logger.warn({
-              service: "historical",
-              msg: `Factory log filter task failed, retrying... [${task.fromBlock}, ${task.toBlock}] (contract=${task.factory.contractName}, network=${this.network.name}, error=${message})`,
-            });
-            const priority = Number.MAX_SAFE_INTEGER - task.fromBlock;
-            queue.addTask(task, { priority, retry: true });
-            break;
-          }
-          case "BLOCK": {
-            this.common.logger.warn({
-              service: "historical",
-              msg: `Block task failed, retrying... [${task.blockNumber}] (network=${this.network.name}, error=${message})`,
-            });
-            const priority = Number.MAX_SAFE_INTEGER - task.blockNumber;
-            queue.addTask(task, { priority, retry: true });
-            break;
-          }
-        }
-      },
-    });
+  private runTask = async (task: HistoricalSyncTask) => {
+    try {
+      if (task.kind === "LOG_FILTER") await this.logFilterTaskWorker(task);
+      else if (task.kind === "FACTORY_LOG_FILTER")
+        await this.factoryLogFilterTaskWorker(task);
+      else if (task.kind === "FACTORY_CHILD_ADDRESS")
+        await this.factoryChildAddressTaskWorker(task);
+      else if (task.kind === "BLOCK") await this.blockTaskWorker(task);
+    } catch (error_) {
+      if (error_ === undefined) return;
```

Comment:
When would this happen?

Reply:
When killing the requestQueue, all promises are immediately rejected, leading to the error being undefined.